include/GoogleSync/GoogleSyncBase.php

```diff
@@ -0,0 +1,1049 @@
+            // The authconfig json string is invalid json
+            // TODO: It should be an exception (do not leave same 'false' return value for differenc failure reason, Inform the caller about this problem)
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'AuthConfig is not proper JSON string');
+            return false;
+        } elseif (!array_key_exists('web', $authJson_local)) {
+            // The authconfig is valid json, but the 'web' key is missing. This is not a valid authconfig.
+            // TODO: It should be an exception (do not leave same 'false' return value for differenc failure reason, Inform the caller about this problem)
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'AuthConfig is missing required [web] key');
+            return false;
+        }
+        return $authJson_local;
+    }
+
+    /**
+     * Creates and Sets the Google client in the object
+     *
+     * @param string $id : the SuiteCRM user id
+     *
+     * @return bool Success/Failure
+     * @throws GoogleSyncException
+     */
+    protected function setClient($id)
+    {
+        $isValidator = new SuiteValidator();
+        if (!$isValidator->isValidId($id)) {
+            throw new GoogleSyncException('Google Sync Base trying to set Client but given an invalid ID: ' . $id, GoogleSyncException::INVALID_CLIENT_ID);
+        }
+        
+        if (!$gClient_local = $this->getClient($id)) {
+            return false;
+        }
+        $this->gClient = $gClient_local;
+        return true;
+    }
+
+    /**
+     * Set the Google client up for the user by id
+     *
+     * @param string $id : the SuiteCRM user id
+     *
+     * @return \Google_Client|false Google_Client on success. False on failure.
+     * @throws GoogleSyncException if unable to retrive the user
+     */
+    protected function getClient($id)
+    {
+        $isValidator = new SuiteValidator();
+        if (!$isValidator->isValidId($id)) {
+            throw new GoogleSyncException('GoogleSyncBase trying to get Client but given ID is invalid: ' . $id, GoogleSyncException::INVALID_CLIENT_ID);
+        }
+
+        // Retrieve user bean
+        if (!isset($this->workingUser)) {
+            $this->workingUser = BeanFactory::getBean('Users', $id);
+            if (!$this->workingUser) {
+                throw new GoogleSyncException('Unable to retrieve a User bean', GoogleSyncException::UNABLE_TO_RETRIEVE_USER);
+            }
+        }
+
+        // Retrieve Access Token JSON from user preference
+        $accessToken = json_decode(base64_decode($this->workingUser->getPreference('GoogleApiToken', 'GoogleSync')), true);
+        if (!array_key_exists('access_token', $accessToken)) {
+            // The Token is invalid JSON or missing
+            // TODO: It should be an exception (do not leave same 'false' return value for differenc failure reason, Inform the caller about this problem)
```

Comment:
@gymad 
I'll look into this. The return is passed to setClient, and then initUserService, which then throws an exception if it failed.
I'll try to figure out a cleaner way to handle it.

Reply:
Done. I'm going to need to do some extensive real world tests to make sure this didn't break anything.